kotlin/query-features-with-arcade-expression/src/main/java/com/esri/arcgisruntime/sample/queryfeatureswitharcadeexpression/MainActivity.kt

```diff
@@ -0,0 +1,196 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgisruntime.sample.queryfeatureswitharcadeexpression
+
+import android.graphics.Color
+import android.graphics.Point
+import android.os.Bundle
+import android.util.Log
+import android.view.MotionEvent
+import android.view.View
+import android.widget.ProgressBar
+import android.widget.TextView
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import com.esri.arcgisruntime.arcade.ArcadeEvaluationResult
+import com.esri.arcgisruntime.arcade.ArcadeEvaluator
+import com.esri.arcgisruntime.arcade.ArcadeExpression
+import com.esri.arcgisruntime.arcade.ArcadeProfile
+import com.esri.arcgisruntime.data.ArcGISFeature
+import com.esri.arcgisruntime.layers.Layer
+import com.esri.arcgisruntime.loadable.LoadStatus
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.view.DefaultMapViewOnTouchListener
+import com.esri.arcgisruntime.mapping.view.MapView
+import com.esri.arcgisruntime.portal.Portal
+import com.esri.arcgisruntime.portal.PortalItem
+import com.esri.arcgisruntime.sample.queryfeatureswitharcadeexpression.databinding.ActivityMainBinding
+import kotlin.math.roundToInt
+
+
+class MainActivity : AppCompatActivity() {
+
+    private val TAG = MainActivity::class.java.simpleName
+
+    // hold a reference to the layer for use in event handlers
+    private var layer: Layer? = null
+
+    // hold a reference to the feature for use in event handlers
+    private var previousFeature: ArcGISFeature? = null
+
+
+    private val activityMainBinding by lazy {
+        ActivityMainBinding.inflate(layoutInflater)
+    }
+
+    private val mapView: MapView by lazy {
+        activityMainBinding.mapView
+    }
+
+    private val progressBar: ProgressBar by lazy {
+        activityMainBinding.progressBar
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(activityMainBinding.root)
+
+        // load the portal and create a map from the portal item
+        val portal = Portal("https://www.arcgis.com/", false)
+        val portalItem = PortalItem(portal, "14562fced3474190b52d315bc19127f6")
+
+        mapView.apply {
+            // set the map to be displayed in the layout's MapView
+            map = ArcGISMap(portalItem)
+            // set the RPD Beats layer to be visible when map is loaded
+            map.addDoneLoadingListener {
+                if (map.loadStatus == LoadStatus.LOADED) {
+                    // set the visibility of all but the RDT Beats layer to false to avoid cluttering the UI
+                    map.operationalLayers.forEach { layer ->
+                        layer.isVisible = layer.name == "RPD Beats  - City_Beats_Border_1128-4500"
+                    }
```

Comment:
I think it'd be perhaps more readable to do

map.operationalLayers.filter { it.name != "RPD Beats... } {
    it.visible = false
}

my syntax questionable here :P

Reply:
We would need to use `forEach` if using `filter{}` or not. I would argue it might be readable to leave it as is.  
### Using filter: 
```kotlin
// set the visibility of all but the RDT Beats layer to false to avoid cluttering the UI
map.operationalLayers.filter { it.name != "RPD Beats  - City_Beats_Border_1128-4500" }.forEach { layer ->
    layer.isVisible = false
}
```
### Using for-each:
```kotlin
// set the visibility of all but the RDT Beats layer to false to avoid cluttering the UI
map.operationalLayers.forEach { layer ->
    layer.isVisible = (layer.name == "RPD Beats  - City_Beats_Border_1128-4500")
}
```